### PR TITLE
[WEB-3979] 14.5.2 - enhanced typings for icons and colours, updated neutral colours

### PR DESCRIPTION
**Pull request by jamiehenson**
## Jira Ticket Link / Motivation

The typing work is a side-chore (WEB-3979), the colour updates are https://ably.atlassian.net/browse/WEB-3987

## Summary of changes

The colour updates are fairly simple, `neutral-100` to `neutral-500` have been updated with new values.

The typing work consists of generating union types for all the possible icon names and Tailwind colour classes, so that we can add further guards around props that use them - this PR adds them as such. 

For example, on the `Icon` component, the value we pass into `name` can no longer just be a string, it has to match a list of icons that we actually have - this prevents accidental errors and highlights gaps in `ably-ui` if we need to add new icons.

## How do you manually test this?

It's a behind the scenes change, so pull it down and mess with some icon name and colour props, I guess.

## Merge/Deploy Checklist

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

### CODE REVIEW

File: src/core/styles/colors/Colors.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import React from "react";
```
Comment (jamiehenson) [inline src/core/styles/colors/Colors.stories.tsx:1]: This file has mainly just been moved

File: src/core/Icon/types.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
+export type IconName =
```
Comment (aralovelace) [inline src/core/Icon/types.ts:1]: This is incredible! Neatly done!

File: src/core/styles/properties.css [hunk 3]
```diff
@@ -2,11 +2,11 @@
   :root {
     /* Neutral colors */
     --color-neutral-000: #ffffff;
-    --color-neutral-100: #f8fafc;
-    --color-neutral-200: #f4f8fb;
-    --color-neutral-300: #edf1f7;
+    --color-neutral-100: #f6f8fa;
+    --color-neutral-200: #eef1f6;
+    --color-neutral-300: #e6eaf0;
```
Comment (aralovelace) [inline src/core/styles/properties.css:7]: Do we need to check the pages that affect these changes? or not a lot of difference between these colors?

**Review by aralovelace — APPROVED** [review]
Nice! just have 1 question

I will just add [this](https://github.com/ably/ably-ui/compare/WEB-3908_icons?expand=1) change once this PR goes live

File: src/core/styles/properties.css (hunk 3 above)
Comment (jamiehenson) [reply src/core/styles/properties.css:7]: Pretty subtle changes - we can just roll this out

**Pull request merged**